components/gcp/dataproc/create_cluster/README.md

```diff
@@ -1,63 +1,88 @@
-
-## Output:
-Name | Description
-:--- | :----------
-cluster_name | The cluster name of the created cluster.
-
-## Sample
+This component can be used at the start of a KFP pipeline to create a temporary Dataproc cluster to run Dataproc jobs as subsequent steps in the pipeline. The cluster can be later recycled by the [Dataproc delete cluster component](https://github.com/kubeflow/pipelines/tree/master/components/gcp/dataproc/delete_cluster).
+
+
+## Runtime arguments
+Name | Description | Type | Optional | Default
+:--- | :---------- | :--- | :------- | :------
+project_id | The Google Cloud Platform (GCP) project ID that the cluster belongs to. | GCPProjectID | No |
+region | The Cloud Dataproc region runs the newly created cluster. | GCPRegion | No |
+name | The name of the newly created cluster. Cluster names within a project must be unique. Names of deleted clusters can be reused. | String | Yes | ` `
+name_prefix | The prefix of the cluster name. | String | Yes | ` `
+initialization_actions | List of Cloud Storage URIs of executables to execute on each node after the configuration is completed. By default, executables are run on the master and all the worker nodes. | List | Yes | `[]`
+config_bucket | A Cloud Storage bucket used to stage the job dependencies, the configuration files, and the job driver console’s output. | GCSPath | Yes | ` `
+image_version | The version of the software inside the cluster. | String | Yes | ` `
+cluster | The full [cluster config] (https://cloud.google.com/dataproc/docs/reference/rest/v1/projects.regions.clusters#Cluster). | Dict | Yes | `{}`
+wait_interval | The number of seconds to pause between polling the operation done status. | Integer | Yes | `30`
+
+## Output
+Name | Description | Type
+:--- | :---------- | :---
+cluster_name | The cluster name of the created cluster. | String
+
+## Cautions & requirements
+To use the component, you must:
+* Setup project by following the [guide](https://cloud.google.com/dataproc/docs/guides/setup-project).
+* The component is running under a secret of [Kubeflow user service account](https://www.kubeflow.org/docs/started/getting-started-gke/#gcp-service-accounts) in a Kubeflow cluster. For example:
+```
+component_op(...).apply(gcp.use_gcp_secret('user-gcp-sa'))
+```
+* Grant Kubeflow user service account the read access to the Cloud Storage buckets which contains initialization action files.
+* Grant Kubeflow user service account the `roles/dataproc.editor` role on the project.
 
-Note: the sample code below works in both IPython notebook or python code directly.
+## Detailed Description
+This component creates a new Dataproc cluster by using [Dataproc create cluster REST API](https://cloud.google.com/dataproc/docs/reference/rest/v1/projects.regions.clusters/create).
 
-### Set sample parameters
+Here are the steps to use the component in a pipeline:
+1. Install KFP SDK
+Install the SDK (Uncomment the code if the SDK is not installed before)
```

Comment:
The codes are not commented here, what do you mean by uncommenting?

Reply:
Good catch. Thanks.